### PR TITLE
close preview listing modal after add to cart -https://particl.atlassian.net/browse/PD-138

### PR DESCRIPTION
**Pull request by vikas-cis**

### CODE REVIEW

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 1]
```diff
@@ -78,6 +78,6 @@
   </div>
   <div class="right" fxLayout fxLayoutAlign="end center">
     <app-favorite [listing]="data?.listing" [detail]="true"></app-favorite>
-    <app-add-to-cart [listing]="data?.listing" [detail]="true"></app-add-to-cart>
+    <app-add-to-cart [listing]="data?.listing" [detail]="true" (onClose)="dialogClose()"></app-add-to-cart>
```
Comment (kewde) [inline src/app/market/listings/preview-listing/preview-listing.component.html]: onAdded

File: src/app/core/market/market.service.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class MarketService {
     return this._http.post(this.url, postData, { headers: headers })
         .map((response: any) => response.result)
         .catch((error: any) => {
-
+          // Inconsistency when you do add to cart
```
Comment (kewde) [inline src/app/core/market/market.service.ts]: More explanation?

Comment (vikas-cis) [reply src/app/core/market/market.service.ts]: we are getting 404 error while try to add same item in the cart and got error something like `error.error.error`, Due to which we got `[object object]` in the error message.

Comment (kewde) [reply src/app/core/market/market.service.ts]: Then fix the error handling..

Comment (kewde) [reply src/app/core/market/market.service.ts]: The in-code comment here is useless.

Comment (vikas-cis) [reply src/app/core/market/market.service.ts]: ok

**Pull request merged**